api/firewaller/machine_test.go

```diff
@@ -161,3 +165,25 @@ func (s *machineSuite) TestIsManual(c *gc.C) {
 	c.Assert(answer, jc.IsTrue)
 
 }
+
+func mustOpenPortRanges(c *gc.C, st *state.State, u *state.Unit, endpointName string, portRanges []network.PortRange) {
+	unitPortRanges, portChangesFn, err := u.OpenedPortRanges()
```

Comment:
Why have two values returned here? The usage of them looks quite disconnected.

Perhaps have the `portChangesFn` actually be a method on the `unitPortRanges`, something like `Operations`?

Reply:
I was just lazy to add this because the correct implementation would need to change the machine_port_ops code to filter the registered open/close port modifications to apply by unit name (otherwise your change set could include the bits for other units as well).

Now that I think about it, it's not _that_ much code so I guess I can implement it today and update the tests as a separate commit.